app/purchasing.cpp

```diff
@@ -398,6 +398,22 @@ void Purchasing::onFetchPurchasingPlansFinished()
 
 void Purchasing::clean()
 {
+  if ( mBackend )
+  {
+    disconnect( mBackend.get(), &PurchasingBackend::planRegistrationSucceeded, this, &Purchasing::onPlanRegistrationSucceeded );
+    disconnect( mBackend.get(), &PurchasingBackend::planRegistrationFailed, this, &Purchasing::onPlanRegistrationFailed );
+
+    disconnect( mBackend.get(), &PurchasingBackend::transactionCreationSucceeded, this, &Purchasing::onTransactionCreationSucceeded );
+    disconnect( mBackend.get(), &PurchasingBackend::transactionCreationFailed, this, &Purchasing::onTransactionCreationFailed );
+  }
```

Comment:
If we delete mBackend, all connections would be removed, so we should not need to disconnect them explicitly